File: 2022-2023/sourcecode/lab/lab6/TaskyJava/app/src/main/java/pl/udu/uwr/pum/taskyjava/TaskyFragment.java
package pl.udu.uwr.pum.taskyjava;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import pl.udu.uwr.pum.taskyjava.adapter.TaskAdapter;
import pl.udu.uwr.pum.taskyjava.data.DataProvider;
import pl.udu.uwr.pum.taskyjava.databinding.FragmentTaskyBinding;
import pl.udu.uwr.pum.taskyjava.model.Task;
import pl.udu.uwr.pum.taskyjava.model.TaskGroup;
import pl.udu.uwr.pum.taskyjava.util.SharedPrefUtil;

public class TaskyFragment extends Fragment {

    private FragmentTaskyBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentTaskyBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        ArrayList<Task> tasks = SharedPrefUtil.getTaskList(requireContext());

        binding.rvTasky.setLayoutManager(new LinearLayoutManager(requireContext()));
        binding.rvTasky.setAdapter(new TaskAdapter(tasks));

        binding.addButton.setOnClickListener(v -> {
            String task = Objects.requireNonNull(binding.taskEditText.getText()).toString();
            String cat = Objects.requireNonNull(binding.groupEditText.getText()).toString();

            if (!task.isEmpty() && !cat.isEmpty()){
                TaskAdapter adapter = (TaskAdapter) Objects.requireNonNull(binding.rvTasky.getAdapter());
                adapter.add(new Task(task, new TaskGroup(cat)), requireContext());
                binding.groupEditText.getText().clear();
                binding.taskEditText.getText().clear();
            }
        });

        binding.clearButton.setOnClickListener(v -> {
            TaskAdapter adapter = (TaskAdapter) Objects.requireNonNull(binding.rvTasky.getAdapter());
            adapter.clear(requireContext());
        });
    }
}

File: 2022-2023/sourcecode/lab/lab6/TaskyKotlin/app/src/main/java/pl/udu/uwr/pum/taskykotlin/TaskyFragment.kt
package pl.udu.uwr.pum.taskykotlin

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import pl.udu.uwr.pum.taskykotlin.adapter.TaskAdapter
import pl.udu.uwr.pum.taskykotlin.data.DataProvider
import pl.udu.uwr.pum.taskykotlin.databinding.FragmentTaskyBinding
import pl.udu.uwr.pum.taskykotlin.model.Task
import pl.udu.uwr.pum.taskykotlin.model.TaskGroup
import pl.udu.uwr.pum.taskykotlin.model.TaskRow
import pl.udu.uwr.pum.taskykotlin.util.getTasksList

class TaskyFragment : Fragment() {

    private lateinit var binding: FragmentTaskyBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentTaskyBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val tasks: MutableList<Task> = getTasksList(requireContext()).toMutableList()
        binding.rvTasky.apply {
            layoutManager = LinearLayoutManager(context)
            adapter = TaskAdapter(tasks)
        }

        binding.addButton.setOnClickListener {
            val task = binding.taskEditText.text.toString()
            val cat = binding.groupEditText.text.toString()

            if (task.isNotEmpty() && cat.isNotEmpty()) {
                (binding.rvTasky.adapter as TaskAdapter).add(Task(task, TaskGroup(cat)), requireContext())
                binding.groupEditText.text?.clear()
                binding.taskEditText.text?.clear()
            }
        }

        binding.clearButton.setOnClickListener {
            (binding.rvTasky.adapter as TaskAdapter).clear(requireContext())
        }
    }
}